react-app/src/components/molecules/Dialog/DialogComponent.tsx

```diff
@@ -13,11 +13,17 @@ import {
 import EditMovieForm from '../Forms/EditMovieForm';
 import AddMovieForm from '../Forms/AddMovieForm';
 import DeleteMovieForm from '../Forms/DeleteMovieForm';
+import { IMovie } from '../MovieCard/IMovie';
 
+// PATTERN: high order component
```

Comment:
I may be missing something, but below doesn't seem to be a good example of HOC. I'd recommend to refer to [official doc](https://reactjs.org/docs/higher-order-components.html) on that topic.